File: apps/backend/src/EnterpriseAgentOs.Domain/features/Management/Interfaces/IUserRepository.cs
namespace EnterpriseAgentOs.Domain.Features.Management;

public interface IUserRepository
{
    Task<UserRecord> UpsertByGoogleSubjectAsync(string googleSubjectId, string email, string? name, string? avatarUrl, CancellationToken ct = default);
    Task<UserRecord> UpsertByGitHubSubjectAsync(string gitHubSubjectId, string email, string? name, string? avatarUrl, CancellationToken ct = default);
    Task<UserRecord?> GetByIdAsync(Guid id, CancellationToken ct = default);
    Task<UserRecord> UpdateProfileAsync(
        Guid id,
        string? name,
        string? displayName,
        string? timezone,
        string? notificationPrefsJson,
        string? preferences,
        CancellationToken ct = default);
    Task DeleteAsync(Guid id, CancellationToken ct = default);
}

File: apps/backend-java/src/main/java/com/enterpriseagentos/backendjava/domain/features/management/IUserRepository.java
package com.enterpriseagentos.backendjava.domain.features.management;

import java.math.BigDecimal;
import java.time.Instant;
import java.time.YearMonth;
import java.util.*;
import java.util.concurrent.CompletableFuture;
import com.enterpriseagentos.backendjava.domain.common.primitives.*;
import com.enterpriseagentos.backendjava.domain.common.services.*;
import com.enterpriseagentos.backendjava.domain.common.valueobjects.*;
import com.enterpriseagentos.backendjava.domain.events.*;
import com.enterpriseagentos.backendjava.domain.features.agents.*;
import com.enterpriseagentos.backendjava.domain.features.analytics.*;
import com.enterpriseagentos.backendjava.domain.features.management.*;
import com.enterpriseagentos.backendjava.domain.features.mcp.*;

public interface IUserRepository {
    CompletableFuture<UserRecord> upsertByGoogleSubjectAsync(String googleSubjectId, String email, String name, String avatarUrl);
    CompletableFuture<UserRecord> upsertByGitHubSubjectAsync(String gitHubSubjectId, String email, String name, String avatarUrl);
    CompletableFuture<UserRecord> getByIdAsync(UUID id);
    CompletableFuture<Void> deleteAsync(UUID id);
}
